test/tests/jobs/FailedJobReplyTest.cpp

```diff
@@ -0,0 +1,49 @@
+#include <test/lib/BedrockTester.h>
+
+struct FailedJobReplyTest : tpunit::TestFixture {
+    FailedJobReplyTest()
+        : tpunit::TestFixture("FailedJobReply",
+                              BEFORE_CLASS(FailedJobReplyTest::setupClass),
+                              TEST(FailedJobReplyTest::failSendingResponse),
+                              AFTER(FailedJobReplyTest::tearDown),
+                              AFTER_CLASS(FailedJobReplyTest::tearDownClass)) { }
+
+    BedrockTester* tester;
+
+    void setupClass() { tester = new BedrockTester(_threadID, {{"-plugins", "Jobs,DB"}}, {});}
+
+    // Reset the jobs table
+    void tearDown() {
+        SData command("Query");
+        command["query"] = "DELETE FROM jobs WHERE jobID > 0;";
+        tester->executeWaitVerifyContent(command);
+    }
+
+    void tearDownClass() { delete tester; }
+
+    // Cannot cancel a job with children
+    void failSendingResponse() {
```

Comment:
I think we should add a test for jobs with retryAfter too.

Reply:
Yeah, there are several more tests to add.